packages/contracts-bedrock/src/dispute/BlockHashOracle.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "../libraries/DisputeTypes.sol";
+import "../libraries/DisputeErrors.sol";
+
+/// @title BlockHashOracle
```

Comment:
do you think its worth emitting an event with blocknum + msg.sender? Would make it easier to track who is providing this service, because providing this service is def a prime for retro funding

Reply:
Yeah this could be neat. We don't need the mapping to be complete, sparse is ok - the idea is to have people that are creating dispute games first store the block hash that they'll be running cannon from here, run cannon, and then create the game with the l1 block number set to the number associated with the block hash.

Reply:
Got it, makes sense